clustermesh-apiserver/clustermesh/synchronizer.go

```diff
@@ -13,25 +13,43 @@ import (
 	"github.com/cilium/hive/cell"
 	"github.com/cilium/hive/job"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
 
 	"github.com/cilium/cilium/clustermesh-apiserver/syncstate"
+	cmnamespace "github.com/cilium/cilium/pkg/clustermesh/namespace"
 	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	k8sConst "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
+	cilium_api_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	cilium_api_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/k8s/types"
 	"github.com/cilium/cilium/pkg/kvstore"
 	"github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
+var (
+	podPrefixLbl = labels.LabelSourceK8s + ":" + k8sConst.PodNamespaceLabel
+)
+
 // Options represents the options to synchronize a given resource type.
 type Options[T runtime.Object] struct {
 	Enabled   bool
 	Resource  string
 	Prefix    string
 	StoreOpts []store.WSSOpt
+	// NamespaceSyncRequired indicates whether namespace changes should trigger resynchronization
+	// of all resources of this type. If true, a namespace watcher will be started to monitor
+	// namespace changes and resynchronize resources accordingly. Only required for certain resource types.
+	NamespaceSyncRequired bool
```

Comment:
`Namespaced` looks better to me (all resources that are namespaced are subject to the new logic).